app/views/map/_leaflet.html.erb

```diff
@@ -1,4 +1,20 @@
-<div style="width:100%;margin-left:0;height:300px;" id="map"></div>
+<style>
+#map { width:100%; height:300px; margin: 0; position: relative;}
```

Comment:
This looks good, but actually one more change -- do you think we could change the id here to something unique, so that multiple maps could in theory appear on a page without conflicting? We could generate a unique id in Ruby using:

```
<%= unique_id = rand(10) %>
```

Then instead of `#map` we'd use:

```
#map<%= unique_id %>
```

Make sense? And we'd do this addition below too on line 19. That way each map instance will be uniquely set up and controlled, even if there are multiple maps on the page.